// Copyright 2022-2024 ETSI SDG TeraFlowSDN (TFS) (https://tfs.etsi.org/)
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// protocol buffers documentation: https://developers.google.com/protocol-buffers/docs/proto3
syntax = "proto3";
package orchestrator;

import "context.proto";


service E2EOrchestratorService {
  rpc Compute(E2EOrchestratorRequest) returns (E2EOrchestratorReply)  {}
  rpc PushTopology(context.Topology)  returns (context.Empty)         {}
}

message E2EOrchestratorRequest {
  context.Service service = 1;
}

message E2EOrchestratorReply {
  // Service requested completed with possible missing fields, and
  // sub-services required for supporting requested service on the
  // underlying layers.
  repeated context.Service services = 1;

  // Connections supporting the requested service and sub-services
  // required for the underlying layers.
  repeated context.Connection connections = 2;
}